solutions/gold/coci-17-sirni.mdx

```diff
@@ -0,0 +1,135 @@
+---
+id: coci-17-sirni
+source: COCI 2017
+title: Sirni
+author: Kevin Sheng
+---
+
+<Spoiler title="Hint 1">
+
+With $10^5$ cards, there's way too many possible edges brute force would have to consider.
+Is there any way we can cut out a majority of these possible edges?
+
+</Spoiler>
+
+<Spoiler title="Hint 2">
+
+Say you have a card with value $5$, and you have two other cards with values $12$ and $13$.
+Is it *ever* optimal to link $5$ to $13$?
+
+</Spoiler>
+
+<Spoiler title="Solution">
+
+## Explanation
+
+Following the first hint, let's try to build up a candidate list of edges that have some semblance of a chance at being included in Daniel's MST.
+
+First, let's sort the cards.
+We can also just not consider any duplicate cards, as those can be linked to other cards of the same value at zero cost.
+
+Now for each card $c$, we want to find all possible cards that might link up with it in an MST.
+Since edges go both ways, we only have to consider cards with values greater than $c$.
+
+Here's where the second hint comes in.
+No matter what other cards are in this example, it will always be optimal for $5$ to link up with $12$ instead of $13$.
```

Comment:
i may be misunderstanding this but you're saying like linking 5 - 12 - 13 is always more optimal than linking 5 - 13 - 12 right

Reply:
i believe so